### ISSUE TITLE
get collections without using a query

### ISSUE THREAD

**Issue by alexchong-psg** [finsweet/fireworkers#13]
Hi,

I have the need to query all of the items in a collection or sub collection.  `Firestore.get` only returns a single document, but get can also return an array of documents.

I ended up forking your repo and adding a getCollection to add a loop when parsing the json results.

I'm happy to make a pull request if you think it would be useful addition.

```
/* eslint-disable @typescript-eslint/no-explicit-any */
import { extract_fields_from_document } from "./fields";
import type * as Firestore from "./types";
import { get_firestore_endpoint } from "./utils";

/**
 * Gets a collection of documents from Firestore.
 * Reference: {@link https://firebase.google.com/docs/firestore/reference/rest/v1/projects.databases.documents/get}
 *
 * @param firestore The DB instance.
 * @param collection_path The collection path.
 */
export const getCollection = async <Fields extends Record<string, any>>(
  { jwt, project_id }: Firestore.DB,
  ...args: string[]
) => {
  const endpoint = get_firestore_endpoint(project_id);
  const collection_path = args.join("/");

  const response = await fetch(`${endpoint}/${collection_path}`, {
    headers: {
      Authorization: `Bearer ${jwt}`,
    },
  });

  const data: Firestore.GetCollectionResponse = await response.json();

  if ("error" in data) throw new Error(data.error.message);

  const documents = data.documents.map((doc) =>
    extract_fields_from_document<Fields>(doc),
  );

  return documents;
};

```

**Comment by alexiglesias93** [comment]
Hey @alexchong-psg !
Have you tried the `query` method?

```typescript
const example = await Firestore.query(db, {
  from: [
    {
      collectionId: 'my_collection_path',
    },
  ],
});
```

I didn't implement a method for retrieving multiple documents because I thought that the `query` method was enough, but maybe you have some use case I am missing?

**Comment by alexchong-psg** [comment]
True, you usually wouldn't want to get all items in a collection without filtering at the root level.

I did try it out, but in order to get a sub-collection, the endpoint would need to change, and it looks like that's hard coded into `query`.  I was able to get the sub-collection in postman with something like below:

`POST https://firestore.googleapis.com/v1/projects/<project>/databases/(default)/documents/users/1234:runQuery`
```
{
    "structuredQuery": {
        "from": [
            {
                "collectionId": "messages",
                "allDescendants": true
            }
        ]
    }
}
```

**Comment by aliyome** [comment]
Hi @alexiglesias93!

I've run into the same issue as well.
The current `query` method seems to be insufficient when wanting to fetch all documents from a sub-collection.
As @alexchong-psg mentioned, we would need to modify the endpoint slightly to query against a sub-collection.
Would it be alright if I create a pull request?

**Comment by aliyome** [comment]
I believe this issue is resolved by PR #14 and #17.

**Comment by alexiglesias93** [comment]
Not entirely because @alexchong-psg asked for a new method to query entire lists instead of single documents with `get`.
But I think it's not necessary as `query` already serves this purpose.

So I guess that yes, we can close this 👍
